email-ingestion/src/email_ingester.rs

```diff
@@ -0,0 +1,206 @@
+use interfaces::defs::{LiveSourceSpec, Ingester, InputItem, WatchRest};
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+
+#[cfg(not(test))]
+use std::sync::Arc;
+
+#[cfg(test)]
+use std::sync::Arc;
+
+pub use crate::tests::mock_server::{MockEmailServer, MockEmail};
+
+#[derive(Clone)]
+pub struct EmailIngesterConfig {
+    pub server: String,
+    pub port: u16,
+    pub username: String,
+    pub password: String,
+    pub mailbox: String,
+    pub use_tls: bool,
+    pub last_sync_date: Option<DateTime<Utc>>,
+}
+
+impl EmailIngesterConfig {
+    pub fn from_env() -> Result<Self> {
+        let last_sync_date = std::env::var("EMAIL_LAST_SYNC")
+            .ok()
+            .and_then(|s| DateTime::parse_from_rfc3339(&s).ok())
+            .map(|dt| dt.with_timezone(&Utc));
+        
+        Ok(Self {
+            server: std::env::var("EMAIL_SERVER")?,
+            port: std::env::var("EMAIL_PORT")?.parse()?,
+            username: std::env::var("EMAIL_USERNAME")?,
+            password: std::env::var("EMAIL_PASSWORD")?,
+            mailbox: std::env::var("EMAIL_MAILBOX").unwrap_or_else(|_| "INBOX".to_string()),
+            use_tls: std::env::var("EMAIL_USE_TLS").unwrap_or_else(|_| "true".to_string()).parse().unwrap_or(true),
+            last_sync_date,
```

Comment:
might be good to add this to the ingester api. would it make sense for other implementations?